foreshadow/transformers/smart.py

```diff
@@ -53,24 +60,39 @@ class Encoder(SmartTransformer):
 
     If there are less than 30 categories, then OneHotEncoder is used, if there are more
     then HashingEncoder is used. If used in a y_var context, LabelEncoder is used.
+    
+        Args:
+            unique_num_cutoff (float): number of allowable unique categories
+            merge_thresh (float): threshold passed into UncommonRemover if selected
 
     """
 
-    def _get_transformer(self, X, y=None, unique_num_cutoff=30, **fit_params):
+    def __init__(self, unique_num_cutoff=30, merge_thresh=0.01, **kwargs):
+        self.unique_num_cutoff = unique_num_cutoff
+        self.merge_thresh = merge_thresh
+        super().__init__(**kwargs)
+
+    def _get_transformer(self, X, y=None, **fit_params):
         data = X.iloc[:, 0]
         col_name = X.columns[0]
         unique_count = len(data.value_counts())
         if self.y_var:
             return LabelEncoder()
-        if unique_count <= unique_num_cutoff:
-            return OneHotEncoder(
-                cols=[col_name],
-                return_df=True,
-                use_cat_names=True,
-                handle_unknown="ignore",
+        if unique_count <= self.unique_num_cutoff:
```

Comment:
What if after the number of categories is less than 30 after merging UncommonRemover?

Reply:
Yep I can make it so that it will resolve after uncommon processes